crates/lib/src/api.rs

```diff
@@ -34,6 +34,11 @@ pub fn compile(
         .map(|p| p.to_string(true))
 }
 
+/// Gets the quilc version
+pub fn get_quilc_version(client: &Qcs) -> Result<String, Box<dyn std::error::Error>> {
```

Comment:
The compiler doesn't like that:

![image](https://user-images.githubusercontent.com/4324359/199563290-947e3f07-f6fe-4ea5-9f27-061335880ed9.png)

Reply:
If `Sync` is the only thing failing, that's fine to remove. I couldn't remember if it's something usually used here or not -- the more important ones are `'static` and `Send`.

Reply:
It'll take `Error + 'static` but not ` + Send` or ` + Sync`

Reply:
Michael and I discussed this offline - he found the issue within quil-rs:

```
1. Box has two From impls related to errors: From<E> for Box<dyn Error + 'a> and From<E> for Box<dyn Error + Send + Sync + 'a>. That is, there's no impl for Box<dyn Error + Send + 'a.
2. quilc::Error is not Sync because quil_rs::program::SyntaxError is not Sync.
3. SyntaxError is not Sync because quil_rs::parser::error::error::Error is not Sync
4. quil_rs::parser::error::error::Error is not Sync because it contains a field of type Option<Box<dyn Error + Send + 'static>>.
```

We have to add `Sync` to the parser error in quil-rs, then we can do it here. Adding a TODO since there is another method in this API with the same problem.